__tests__/base.js

```diff
@@ -2090,6 +2090,46 @@ function testObjectTypes(produce) {
 			})
 		})
 	}
+
+	describe("class with getters", () => {
+		class State {
+			[immerable] = true
+			_bar = {baz: 1}
+			foo
+			get bar() {
+				return this._bar
+			}
+			syncFoo() {
+				this.foo = this.bar.baz
+			}
+		}
+		const state = new State()
+
+		it("should use a method to assing a field using a getter that return a non primitive object", () => {
+			const newState = produce(state, draft => {
+				draft.syncFoo()
```

Comment:
could you assert the values of `draft.bar`, `state.bar` and `newState.bar` to be correct as well?